src/Core/ServiceProvider.php

```diff
@@ -0,0 +1,87 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.2.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Core;
+
+use League\Container\ServiceProvider\AbstractServiceProvider;
+use League\Container\ServiceProvider\BootableServiceProviderInterface;
+
+/**
+ * Container ServiceProvider
+ *
+ * Service provider bundle related services together helping
+ * to organize your application's dependencies. They also help
+ * improve performance of applications with many services by
+ * allowing service registration to be deferred until services are needed.
+ *
+ * @experimental This class' interface is not stable and may change
+ *   in future minor releases.
+ */
+abstract class ServiceProvider extends AbstractServiceProvider
+    implements BootableServiceProviderInterface
+{
+    /**
+     * Delegate to the bootstrap() method
+     *
+     * This method primarily exists as a shim between the interface
+     * that league/container has and the one we want to offer in CakePHP.
```

Comment:
Based on this should we mark this method `final` then?

Reply:
Nah, if people want to footgun themselves they can.

Reply:
😝